src/vss_tools/vspec/vssexporters/vss2samm/helpers/ttl_builder_helper.py

```diff
@@ -0,0 +1,653 @@
+                enum_values = None
+                if hasattr(vss_node.data, "allowed") and vss_node.data.allowed and type(vss_node.data.allowed) is list:
+                    # Add allowed values to this node characteristic
+                    enum_values = get_enum_values(vss_node.data.allowed)
+                elif hasattr(vss_node.data, "enum") and vss_node.data.enum:
+                    # NOTE: From VSS 3.0 the 'enum' attribute has been renamed to `allowed`
+                    #       However, we will keep this for backwards compatibility.
+
+                    # Add ENUM values, as usual, to this node characteristic
+                    enum_values = get_enum_values(vss_node.data.enum)  # type: ignore
+
+                if enum_values is not None:
+                    __add_node_tuple(graph, node_char_uri, SammCConcepts.VALUES.uri, enum_values)
+
+            case SammCConcepts.LIST:
+                # Handle LIST type nodes
+                log.debug("  -- set node List values")
+
+                # TODO: do we need to add values here?
+
+            case SammCConcepts.MEASUREMENT:
+                # Handle MEASUREMENT (unit) type nodes
+                log.debug("  -- set node Measurement values")
+
+                if hasattr(vss_node.data, "unit") and vss_node.data.unit:
+                    __add_node_tuple(
+                        graph, node_char_uri, SammCConcepts.UNIT.uri, vss_helper.get_data_unit_uri(vss_node.data.unit)
+                    )
+
+                else:
+                    log.warning(
+                        "Unit is not set for vss node: '%s' with type: '%s'\n"
+                        "Setting node 'RDF.type' from MEASUREMENT to default: CHARACTERISTIC\n",
+                        vss_node.name,
+                        vss_node.data.datatype,  # type: ignore
+                    )
+
+                    rdf_type = SammConcepts.CHARACTERISTIC
+
+                if vss_node.data.default:  # type: ignore
+                    __add_node_tuple(
+                        graph,
+                        node_uri,
+                        SammConcepts.EXAMPLE_VALUE.uri,
+                        Literal(vss_node.data.default, datatype=data_type),  # type: ignore
+                    )
+
+            case SammConcepts.CHARACTERISTIC:
+                # Handle CHARACTERISTIC type nodes
+                log.debug(" -- set regular node values")
+
+                if vss_node.data.default:  # type: ignore
+                    if (
+                        hasattr(vss_node.data, "unit")
+                        and vss_node.data.unit
+                        and vss_node.data.unit == "iso8601"
+                        and data_type == DataTypes[vss_node.data.unit]
+                    ):
+                        # Handle date-time based nodes
+                        log.debug(" -- set node DATE-TIME values")
+
+                        # NOTE: Some examples are provided in the form: 0000-01-01T00:00Z
```

Comment:
This is an interesting mismatch. In ISO 8601-1:2019 section 4.3.2 they state:


_The Gregorian calendar defines a calendar year to be either 365 or 366 days, which begins on January 1
and ends on December 31. Each Gregorian calendar year can be identified by a 4-digit ordinal number
beginning with ‘0000’ for year zero, through ‘9999’._

they also support negative numbers so it supports year -1, 0, 1, and so on. Do you know if ESMF AME supports negative years, like -1? If so I think it would be theoreticaly more correct to use -1 rather than +1 in the samm export (see https://en.wikipedia.org/wiki/Year_zero). But this is quite hypothetical anyway, so no need to change anything.

Reply:
Hello Erik.

I think I mentioned it in the comment for this check. Anyway, It seems that the ESMF SAMM validation does not accept the "zero year". Therefore I had to implement this validation at this stage and make sure that there is no "zero year" in some date related nodes.